src/EnergyPlus/CondenserLoopTowers.cc

```diff
@@ -6463,6 +6455,30 @@ namespace CondenserLoopTowers {
         }
     }
 
+    void CoolingTower::checkMassFlowAndLoad(EnergyPlusData &state, Real64 const MyLoad, bool &returnFlagSet)
+    {
+        // MassFlowTolerance is a parameter to indicate a no flow condition
+        if (this->WaterMassFlowRate <= DataBranchAirLoopPlant::MassFlowTolerance) {
+            // for multiple cells, we assume that it's a common basin
+            CalcBasinHeaterPower(
+                state, this->BasinHeaterPowerFTempDiff, this->BasinHeaterSchedulePtr, this->BasinHeaterSetPointTemp, this->BasinHeaterPower);
+            returnFlagSet = true;
+        }
+
+        // Also check to see if the tower is not running (zero MyLoad)
+        if (std::abs(MyLoad) <= HVAC::SmallLoad) {
+            // tower doesn't need to do anything
+            this->OutletWaterTemp = state.dataLoopNodes->Node(this->WaterInletNodeNum).Temp;
+            this->WaterMassFlowRate = 0.0;
```

Comment:
I think when there is a local change to plant flow rates, that change should be confirmed using this call. The point is, after the water flow is set to 0 here, there should be a call to the plant component flow control function SetComponentFlowRate. It is possible that the plant flow resolver would want flow through this branch (i.e., maybe there is no bypass and the pump is constant volume).

        PlantUtilities::SetComponentFlowRate(state, this->WaterMassFlowRate, this->WaterInletNodeNum, this->WaterOutletNodeNum, this->plantLoc);

Reply:
Should the order of these conditionals be switched? line 6461 and line 6472. Why do this twice? This begs an else if.